solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -287,7 +288,14 @@ public Object call() throws Exception {
     }
   }
 
-  public ZkController(final CoreContainer cc, String zkServerAddress, int zkClientConnectTimeout, CloudConfig cloudConfig, final CurrentCoreDescriptorProvider registerOnReconnect)
+  /**
+   * @param cc Core container associated with this controller. cannot be null.
+   * @param zkServerAddress where to connect to the zk server
+   * @param zkClientConnectTimeout timeout in ms
+   * @param cloudConfig configuration for this controller. TODO: possibly redundant with CoreContainer
+   * @param registerOnReconnect a provider of the current core descriptors. used to know what to reregister after a reconnect
```

Comment:
I wonder if "registerOnReconnect" is not appropriate for the name.  It's named how it's being used but not for what it is.  Imagine how silly if all things were named with such an approach.  Perhaps simply "descriptorProvider".  Any way, leave or change as you wish.